frontstage/views/surveys.py

```diff
@@ -1,181 +1,159 @@
+import json
 import logging
-from os import getenv
 
-import arrow
-from flask import Blueprint, render_template, request, url_for
-import requests
+from flask import Blueprint, redirect, render_template, request, url_for
 from structlog import wrap_logger
-from werkzeug.utils import redirect
 
 from frontstage import app
-from frontstage.common.post_event import post_event
+from frontstage.common.api_call import api_call
 from frontstage.common.authorisation import jwt_authorization
-from frontstage.exceptions.exceptions import ExternalServiceError
+from frontstage.exceptions.exceptions import ApiError
 
 
 logger = wrap_logger(logging.getLogger(__name__))
 
 surveys_bp = Blueprint('surveys_bp', __name__, static_folder='static', template_folder='templates/surveys')
 
 
-# ===== Surveys To Do =====
 @surveys_bp.route('/', methods=['GET'])
 @jwt_authorization(request)
 def logged_in(session):
-    status_filter = ["not started", "downloaded"]
-    data_array = build_surveys_data(session, status_filter)
-    return render_template('surveys/surveys-todo.html', _theme='default', data_array=data_array)
+    party_id = session.get('party_id')
+    logger.info('Retrieving surveys todo list', party_id=party_id)
+    surveys_list = get_surveys_list(party_id, 'todo')
+    logger.info('Successfully retrieved surveys todo list', party_id=party_id)
+    return render_template('surveys/surveys-todo.html', _theme='default', surveys_list=surveys_list)
 
 
-# ===== Surveys History =====
 @surveys_bp.route('/history', methods=['GET'])
 @jwt_authorization(request)
 def surveys_history(session):
-    status_filter = ["complete"]
-    data_array = build_surveys_data(session, status_filter)
-    return render_template('surveys/surveys-history.html',  _theme='default', data_array=data_array, history=True)
+    party_id = session['party_id']
+    logger.info('Retrieving surveys history list', party_id=party_id)
+    surveys_list = get_surveys_list(party_id, 'history')
+    logger.info('Successfully retrieved surveys history list', party_id=party_id)
+    return render_template('surveys/surveys-history.html',  _theme='default', surveys_list=surveys_list, history=True)
+
+
+def get_surveys_list(party_id, list_type):
+    logger.debug('Retrieving surveys list', party_id=party_id, list_type=list_type)
+    params = {
+        "party_id": party_id,
+        "list": list_type
+    }
+    response = api_call('GET', app.config['SURVEYS_LIST'], parameters=params)
+    if response.status_code != 200:
+        logger.error('Failed to retrieve surveys list', party_id=party_id, list_type=list_type)
+        raise ApiError(response)
+    surveys_list = json.loads(response.text)
+    logger.debug('Successfully retrieved surveys list', party_id=party_id, list_type=list_type)
```

Comment:
Do you need this log line here in debug and above in info?

Reply:
Actually your right for this function it doesn't make sense, I'll update it